presto-native-execution/presto_cpp/main/PrestoExchangeSource.cpp

```diff
@@ -198,7 +200,8 @@ void PrestoExchangeSource::doRequest(
       .thenValue([path, maxBytes, maxWait, self](
                      std::unique_ptr<http::HttpResponse> response) {
         velox::common::testutil::TestValue::adjust(
-            "facebook::presto::PrestoExchangeSource::doRequest", self.get());
+            "facebook::presto::PrestoExchangeSource::doRequest::thenValue",
+            self.get());
         auto* headers = response->headers();
         if (headers->getStatusCode() != http::kHttpOk &&
```

Comment:
Could you help refactor code by putting this in a member function say handle PrestoExchangeSource::handleResponse? Then call into different utilities based on the actual error types? Thanks!